test/features/router/misc.js

```diff
@@ -84,4 +86,89 @@ describe('router - misc', function() {
         });
     });
 
+    it('should correctly resolve request templates', function() {
+        var requestTemplate = {
+            uri: '/{domain}/test',
+            method: 'post',
+            headers: {
+                'name-with-dashes': '{name-with-dashes}',
+                'global-header': '{$.request.params.domain}',
+                'added-string-header': 'added-string-header'
+            },
+            query: {
+                'simple': '{simple}',
+                'added': 'addedValue',
+                'global': '{$.request.headers.name-with-dashes}'
+            },
+            body: {
+                'object': '{object}',
+                'global': '{$.request.params.domain}',
+                'added': 'addedValue',
+                'nested': {
+                    'one': {
+                        'two': {
+                            'tree': '{a.b.c}'
```

Comment:
@d00rman It has, here it's level 4 (note nested), and in body it's level 3 (no nested)